### PR TITLE
Iox #590 plantuml in design documentation

### PR DESCRIPTION
**Pull request by dkroenke**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer
This PR introduces a guideline for using plantuml in the design documentation. It describes how to integrate plantuml into the markdown files.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- relates to #590

### CODE REVIEW

**Review by elBoberido — DISMISSED** [review]
LGTM

File: doc/design/README.md [hunk 1]
```diff
@@ -1,8 +1,39 @@
...
+In here you can find design documents to all constructs of iceoryx. At the
 moment those design documents are neither complete nor final.
 
-In [draft](./draft/) are design drafts of current features contained which are under 
-implementation. The goal is to provide other developers a first idea on how 
+In [draft](./draft/) are design drafts of current features contained which are under
+implementation. The goal is to provide other developers a first idea on how
 the feature can be used and how it will be implemented.
+
+## Add diagrams using plantuml
```
Comment (mossmaurice) [inline doc/design/README.md:10]: Wikipedia writes it as [PlantUML](https://en.wikipedia.org/wiki/PlantUML). You could think of using the same style.

**Review by mossmaurice — COMMENTED** [review]
LGTM

File: doc/design/diagrams/puml_github_workflow.svg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" contentScriptType="application/ecmascript" contentStyleType="text/css" height="405px" preserveAspectRatio="none" style="width:1007px;height:405px;background:#EEEBDC;" version="1.1" viewBox="0 0 1007 405" width="1007px" zoomAndPan="magnify"><defs><filter height="300%" id="f18ey02a78uejz" width="300%" x="-1" y="-1"><feGaussianBlur result="blurOut" stdDeviation="2.0"/><feColorMatrix in="blurOut" result="blurOut2" type="matrix" values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 .4 0"/><feOffset dx="4.0" dy="4.0" in="blurOut2" result="blurOut3"/><feBlend in="SourceGraphic" in2="blurOut3" mode="normal"/></filter></defs><g><!--MD5=[81da3b3e20e7b00bc69cac0acd70a916]
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply doc/design/diagrams/puml_github_workflow.svg:1]: Ok, yeah that's an issue. What about generating the SVG just for releases and then adding the release tag in file name?

[the remark this answers is not in the document]
Comment (elBoberido) [reply doc/design/diagrams/puml_github_workflow.svg:1]: Well, it's quite cumbersome to change all the paths for every release. I would keep the pragmatic solution. There is also a script which keeps everything in sync.

**Review by mossmaurice — APPROVED** [review]

**Review by elBoberido — APPROVED** [review]

**Pull request merged**